instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/snsattributes.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelaws // import "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/aws/smithy-go/middleware"
+
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.27.0"
+)
+
+// SNSAttributeSetter sets SNS specific attributes depending on the SNS operation is being performed.
+func SNSAttributeSetter(ctx context.Context, in middleware.InitializeInput) []attribute.KeyValue {
+	snsAttributes := []attribute.KeyValue{semconv.MessagingSystemKey.String("aws_sns")}
+
+	switch v := in.Parameters.(type) {
+	case *sns.PublishBatchInput:
+		snsAttributes = append(snsAttributes,
+			semconv.MessagingDestinationName(extractDestinationName(v.TopicArn, nil, nil)),
+			semconv.MessagingOperationTypePublish,
+			semconv.MessagingOperationName("publish_batch_input"),
+			semconv.MessagingBatchMessageCount(len(v.PublishBatchRequestEntries)),
+		)
+	case *sns.PublishInput:
+		snsAttributes = append(snsAttributes,
+			semconv.MessagingDestinationName(extractDestinationName(v.TopicArn, v.TargetArn, v.PhoneNumber)),
+			semconv.MessagingOperationTypePublish,
+			semconv.MessagingOperationName("publish_input"),
+		)
+	}
+
+	return snsAttributes
+}
+
+func extractDestinationName(topicArn, targetArn, phoneNumber *string) string {
+	if topicArn != nil && *topicArn != "" {
+		return (*topicArn)[strings.LastIndex(*topicArn, ":")+1:]
+	} else if targetArn != nil && *targetArn != "" {
+		return (*targetArn)[strings.LastIndex(*targetArn, ":")+1:]
+	} else if phoneNumber != nil {
```

Comment:
Do we want to make phone number something that we can disable. Maybe allow for the use 'SMS' or something along those lines as a default.

Reply:
Hi @akats7 this will use Phone Number `only` if you don't provide topicArn or targetArn, but if phone number is not present, it will default to an empty string.

Reply:
Btw, used a similar logic to the one in JS https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sns.ts#L49

Reply:
Yep I understand, I'm just thinking in cases when users don't want phone numbers as attributes, either for PII or cardinality reasons, the cardinality reason probably applies to topic as well.

Reply:
I'm on board with being conservative to start and not include the phone number for now. If the semconv group decides otherwise we can add it in then, right?

Reply:
That sounds good to me; I will make a change